src/System.Text.Json.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -10,7 +10,17 @@ namespace System.Text.Json
 {
     public class JsonDynamicObject : DynamicObject, IBufferFormattable
     {
-        Dictionary<JsonProperty, JsonValue> _properties;
+        #region Constants
+
+        private static readonly char[] EncodingChars = new char[] { '{', '}', ',', ':', '"' };
+        private static readonly ReadOnlySpan<char> OpenCurlySpan = new ReadOnlySpan<char>(EncodingChars, 0, 1);
+        private static readonly ReadOnlySpan<char> CloseCurlySpan = new ReadOnlySpan<char>(EncodingChars, 1, 1);
+        private static readonly ReadOnlySpan<char> CommaSpan = new ReadOnlySpan<char>(EncodingChars, 2, 1);
+        private static readonly ReadOnlySpan<char> ColonSpan = new ReadOnlySpan<char>(EncodingChars, 3, 1);
```

Comment:
This would be part of the existing JSON work item. I suspect that there is probably work that needs to be done it we productive this JSON serializer.